### PR TITLE
feat: add v5 implementation of header height context

### PR DESCRIPTION
**Pull request by WoLewicki**
## Description

Added context for header height in v5 native-stack. For now it does not include `largeTitle` option on `iOS`. We might consider changing this. Based on https://github.com/react-navigation/react-navigation/pull/9774.

## Test code and steps to reproduce

`Test1031.tsx`

## Checklist

- [x] Included code example that can be used to test this change
- [x] Ensured that CI passes

### CODE REVIEW

File: TestsExample/ios/TestsExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -728,6 +729,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = "org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)";
 				PRODUCT_NAME = TestsExample;
 				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (WoLewicki) [inline TestsExample/ios/TestsExample.xcodeproj/project.pbxproj:732]: This adds an option to run `TestsExample` as native iPad app

File: src/native-stack/views/NativeStackView.tsx [hunk 2]
```diff
@@ -93,12 +109,31 @@ const MaybeNestedStack = ({
...
+  const dimensions = useSafeAreaFrame();
+  // landscape is meaningful only for iPhone
+  const isLandscape =
+    dimensions.width > dimensions.height &&
+    !(Platform as PlatformIOSStatic).isPad &&
+    !(Platform as PlatformIOSStatic).isTVOS;
+  // `modal` and `formSheet` presentations do not take whole screen, so should not take the inset.
+  const isFullScreenModal =
+    stackPresentation !== 'modal' && stackPresentation !== 'formSheet';
+  const topInset = isFullScreenModal && !isLandscape ? insets.top : 0;
```
Comment (kacperkapusciak) [inline src/native-stack/views/NativeStackView.tsx:122]: These variables could potentially be moved into the `getDefaultHeaderHeight` function

File: TestsExample/src/Test1031.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <Button title="Tap me for second screen" onPress={() => navigation.navigate('Second')} />
+
+  );
+}
+
+function Second({navigation}: {navigation: NativeStackNavigationProp<ParamListBase>}) {
+  console.log(useHeaderHeight());
+  return (
+    <Button title="Tap me for second screen" onPress={() => navigation.navigate('First')} />
```
Comment (Ubax) [inline TestsExample/src/Test1031.tsx:33]: Typo - I think it should be for first screen

File: src/native-stack/views/NativeStackView.tsx (hunk 2 above)
Comment (Ubax) [reply src/native-stack/views/NativeStackView.tsx:122]: I agree with @kacperkapusciak. Could you move them to `getDefaultHeaderHeight` or is there a reason for them being here?

**Pull request merged**